osu.Game/Screens/Tournament/Components/VisualiserLine.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shaders;
+using System;
+using osu.Framework.Graphics.OpenGL;
+using osu.Framework.Graphics.Batches;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Timing;
+using osu.Framework.Graphics.Transforms;
+
+namespace osu.Game.Screens.Tournament.Components
+{
+    class VisualiserLine : Drawable
+    {
+        private float strokeWidth = 1;
+        public float StrokeWidth
+        {
+            get { return strokeWidth; }
+            set
+            {
+                if (strokeWidth == value)
+                    return;
+
+                strokeWidth = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private float strokeHeight = 1;
+        public float StrokeHeight
+        {
+            get { return strokeHeight; }
+            set
+            {
+                if (strokeHeight == value)
+                    return;
+
+                strokeHeight = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private float separation = 0;
+        public float Separation
+        {
+            get { return separation; }
+            set
+            {
+                if (separation == value)
+                    return;
+
+                separation = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private Shader shader;
+
+        private VisualiserLineDrawNodeSharedData visualiserLineDrawNodeSharedData => new VisualiserLineDrawNodeSharedData();
+
+        /// <summary>
+        /// The period of this visualiser line, in radians.
+        /// </summary>
+        private float period;
+
+        /// <summary>
+        /// The period offset this line was constructed with, in radians.
+        /// </summary>
+        private float initialPeriodOffset;
+        /// <summary>
+        /// The rolling period offset (by transformation), in radians.
+        /// </summary>
+        private float internalPeriodOffset;

```

Comment:
i'd probably call this `_periodOffset`. also move the xmldoc to the property which should be used below